google/cloud/internal/future_impl.h

```diff
@@ -204,6 +204,10 @@ class future_shared_state_base {
     continuation_ = std::move(c);
   }
 
+  std::function<void()> extract_cancellation_callback() {
```

Comment:
"release" would be more idiomatic than "extract" methinks.

Reply:
I agree, sending a PR in a few minutes.